src/main/java/net/cosmocat/marketplace/aop/FeatureToggleAspect.java

```diff
@@ -0,0 +1,39 @@
+package net.cosmocat.marketplace.aop;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import net.cosmocat.marketplace.exception.type.FeatureNotAvailableException;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class FeatureToggleAspect {
+
+  private final FeatureToggleService featureToggleService;
+
+  @Before("@annotation(net.cosmocat.marketplace.aop.FeatureToggle)")
+  public void checkFeatureToggle(JoinPoint joinPoint) {
+    MethodSignature signature = (MethodSignature) joinPoint.getSignature();
+    FeatureToggle featureToggle = signature.getMethod().getAnnotation(FeatureToggle.class);
+
+    String featureName = featureToggle.value();
+
+    log.debug("Checking feature toggle for: {}", featureName);
```

Comment:
I like logs, but there is the thing, to see the debug logs - you will need to run app in the debug mode. it's almost never the case, cause than you need to re-deploy with different tags. 

So here the info would be more prefferable :)